R/roxygen-examples-add-remove.R

```diff
@@ -0,0 +1,32 @@
+#' Remove dont* mask
+#'
+#' @param roxygen Roxygen code examples that contains a dont* segment only.
+#' @keywords internal
+remove_dont_mask <- function(roxygen) {
+  potential_pos <- c(3L, length(roxygen) - 1L)
+  is_line_break_at_potential_pos <- which(roxygen[potential_pos] == "\n")
+  mask <- c(
+    1L, 2L, length(roxygen), potential_pos[is_line_break_at_potential_pos]
+  ) %>% sort()
+  list(
+    code = roxygen[-mask], mask = paste(roxygen[seq2(1, 2)], collapse = "")
+  )
+}
+
+remove_blank_lines <- function(code) {
+  code[code != "\n"]
```

Comment:
Are the terminal newlines part of the contents of a character vector? In `readLines()` (and many other functions), each vector element is a line, but the newlines aren't part of it. This design is surprising at least.

Reply:
Does this help get rid of trailing newlines at a very early stage?

``` r
library(magrittr)
tools::parse_Rd(textConnection("\\dontrun{x <- 3\n\n # hi}"), fragment = TRUE)[[1]] %>% as.character()
#> [1] "x <- 3\n" "\n"       " # hi"
```

> Created on 2018-07-15 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

Reply:
Not sure because we must keep the `dontrun`. We just want the following mapping: 
``` r
library(magrittr)
# 1) parse
rd <- tools::parse_Rd(textConnection(
  "\\dontrun{x <- 3\n\n # hi}"
), fragment = TRUE) %>%
  as.character()
rd
#> [1] "\\dontrun" "{"         "x <- 3\n"  "\n"        " # hi"     "}"        
#> [7] "\n"

# 2) map to styler line definition
mapping_to_be_defined(rd)
# > [1] "\\dontrun{x <- 3" "" "# hi}"
```
Also, the problem I think is that `strsplit` does not split if the character is already the last in the string.
``` r
library(magrittr)
c("a\n", "x") %>% strsplit("\n", fixed = TRUE) %>% unlist()
#> [1] "a" "x"
# should be
# > [1] "a" "" "x
```

Created on 2018-07-15 by the [reprex
package](http://reprex.tidyverse.org) (v0.2.0).

Reply:
Not to confuse anywone, just a note for myself. A problem might be that `strsplit` does not split if the character is already the last in the string.
``` r
library(magrittr)
c("a\n", "x") %>% strsplit("\n", fixed = TRUE) %>% unlist()
#> [1] "a" "x"
# should be
# > [1] "a" "" "x
```

Created on 2018-07-15 by the [reprex
package](http://reprex.tidyverse.org) (v0.2.0).